hudi-common/src/main/java/org/apache/hudi/common/util/collection/LazyFileIterable.java

```diff
@@ -38,6 +38,8 @@
   // Stores the key and corresponding value's latest metadata spilled to disk
   private final Map<T, DiskBasedMap.ValueMetadata> inMemoryMetadataOfSpilledData;
 
+  private transient Thread shutdownThread = null;
```

Comment:
@bvaradar Any reason this needs to be transient ? Are we serializing this ?

Reply:
I think I had the same question. Its just defensive coding.